src/pool-bin/libraries/BinPool.sol

```diff
@@ -347,9 +356,13 @@ library BinPool {
 
         /// @dev overflow check on total reserves and the resulting liquidity
         uint256 price = activeId.getPriceFromId(binStep);
-        binReserves.add(amountIn).getLiquidity(price);
+        bytes32 newReserves = binReserves.add(amountIn);
+        uint256 liquidity = newReserves.getLiquidity(price);
+        if (liquidity > Constants.MAX_LIQUIDITY_PER_BIN) {
+            revert BinPool__MaxLiquidityPerBinExceeded();
+        }
 
```

Comment:
should keep same logic with line 474 ?

Reply:
sorry, what do u mean ?

Reply:
never mind , you removed "uint256 liquidity = newReserves.getLiquidity(price);" in latest commit

Reply:
eh, because otherwise forge coverage would fail with stack too deep